tensorflow/lite/experimental/litert/vendors/qualcomm/core/builders/conv2d_op_builder.cc

```diff
@@ -50,30 +52,54 @@ std::vector<OpWrapper> BuildConv2dOp(
     axis_quant_params.SetAxis(new_axis[axis_quant_params.GetAxis()]);
   }
 
-  const std::vector<std::uint32_t> permute_dims{
-      filter_tensor.GetDim(kHeightIndex), filter_tensor.GetDim(kWidthIndex),
-      filter_tensor.GetDim(kChannelIndex), filter_tensor.GetDim(kBatchIndex)};
-  auto& transposed_filter_tensor =
-      tensor_pool.CloneNativeTensorFrom(filter_tensor, permute_dims);
-
-  const std::vector<std::uint32_t> permute_shape{4};
-  const std::array<std::uint32_t, 4> permute_data{kHeightIndex, kWidthIndex,
-                                                  kChannelIndex, kBatchIndex};
-  auto& permute_tensor = tensor_pool.CreateStaticTensor(
-      QNN_DATATYPE_UINT_32, QuantizeParamsWrapperVariant{}, permute_shape,
-      sizeof(decltype(permute_data)::value_type) * permute_data.size(),
-      permute_data.data());
-
-  OpWrapper& transpose_op = CreateOpWrapper(res, QNN_OP_TRANSPOSE);
-  transpose_op.AddInputTensor(filter_tensor);
-  transpose_op.AddOutputTensor(transposed_filter_tensor);
-  transpose_op.AddTensorParam(QNN_OP_TRANSPOSE_PARAM_PERM, permute_tensor);
+  int filters_num = filter_tensor.GetTensorNumElements();
+  size_t filter_bytes = filter_tensor.GetTensorBytes();
+  TensorWrapper* transposed_filter_tensor;
```

Comment:
Nit: always give initial value.
```suggestion
  TensorWrapper* transposed_filter_tensor = nullptr;
```